ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java

```diff
@@ -120,25 +123,47 @@ public boolean apply(@Nullable final DetailedS3Object input) {
         };
     }
 
-    private Predicate<DetailedS3Object> getDatePredicate() throws CommandException {
+    private Predicate<DetailedS3Object> getDatePredicate() throws ParseException {
         if (Guard.isMapNullOrEmpty(this.filterParams)) {
             return Predicates.notNull();
         }
 
+        // newerthan and olderthan are relative
         final String newer = this.filterParams.get(NEWERTHAN);
         final String older = this.filterParams.get(OLDERTHAN);
-        if (Guard.isStringNullOrEmpty(newer) && Guard.isStringNullOrEmpty(older)) {
+        // before and after are actual dates
+        final String after = this.filterParams.get(AFTER);
+        final String before = this.filterParams.get(BEFORE);
+
+        // if any of these are set, create a range for the predicate
+        if (!Guard.isStringNullOrEmpty(newer) && !Guard.isStringNullOrEmpty(older)
+                && !Guard.isStringNullOrEmpty(after) && !Guard.isStringNullOrEmpty(before)) {
             return Predicates.notNull();
         }
-        // if one is specified, default the other
-        final long newerThan = Guard.isStringNullOrEmpty(newer) ? 0L : new Date().getTime() - Utils.dateDiffToSeconds(newer) * MILLIS_PER_SECOND;
-        final long olderThan = Guard.isStringNullOrEmpty(older) ? Long.MAX_VALUE : new Date().getTime() - Utils.dateDiffToSeconds(older) * MILLIS_PER_SECOND;
+
+        long newerThan = 0L;
+        long olderThan = Long.MAX_VALUE;
+
+        if (!Guard.isStringNullOrEmpty(newer)) {
```

Comment:
If one is set the other range element defaults to Date(0) - Date(MAX). If both starts or both ends are set, the absolute date wins.

I emailed George T:
"Deranged use could even mix relative and absolute. This year, older than two days is after:Y2016,olderthan:d2 but if both are provided, (say after and newerthan) absolute would win."